### PR TITLE
feat: added langfuse integration

### PR DESCRIPTION
**Pull request by brenkao**

### CODE REVIEW

File: mirascope/base/utils.py [hunk 1]
```diff
@@ -222,7 +222,8 @@ def wrapper_generator(*args, **kwargs):
         try:
             for attempt in retries:
                 with attempt:
-                    for value in fn(*args, **kwargs):
+                    generator = fn(*args, **kwargs)
```
Comment (willbakst) [inline mirascope/base/utils.py:225]: is this necessary for some reason? if so please add a comment

File: mirascope/logfire/logfire.py [hunk 2]
```diff
@@ -379,19 +379,57 @@ def with_logfire(cls):
         setattr(cls, "add", mirascope_logfire_span(cls.add))
     if get_parent_class_name(cls, "OpenAI"):
         if hasattr(cls, "call_params"):
-            cls.call_params.logfire = logfire.instrument_openai
+            setattr(
```
Comment (willbakst) [inline mirascope/logfire/logfire.py:382]: we likely also want to do this with `weave`?

**Review by willbakst — APPROVED** [review]

File: mirascope/base/utils.py (hunk 1 above)
Comment (brenkao) [reply mirascope/base/utils.py:225]: it was for debugging, switched back

File: mirascope/logfire/logfire.py (hunk 2 above)
Comment (brenkao) [reply mirascope/logfire/logfire.py:382]: done

**Pull request merged**